secure-api-gateway-ob-uk-common-datamodel/src/main/java/com/forgerock/sapi/gateway/ob/uk/common/datamodel/converter/v4/event/FREventSubscriptionConverter.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2020-2024 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.ob.uk.common.datamodel.converter.v4.event;
+
+import com.forgerock.sapi.gateway.ob.uk.common.datamodel.event.FREventSubscriptionData;
+import uk.org.openbanking.datamodel.v4.event.OBEventSubscription1;
+
+public class FREventSubscriptionConverter {
+
+    // OB to FR
+    public static FREventSubscriptionData toFREventSubscriptionData(OBEventSubscription1 obEventSubscription) {
+        return obEventSubscription == null ? null : FREventSubscriptionData.builder()
+                .callbackUrl(obEventSubscription.getData().getCallbackUrl() != null ? obEventSubscription.getData().getCallbackUrl().toString() : null)
+                .version(obEventSubscription.getData().getVersion())
+                .eventTypes(obEventSubscription.getData().getEventTypes())
```

Comment:
Does it matter that it's grabbing the same list from the `obEventSubscription` here? i.e. that the list of `eventTypes` is mutable and could be manipulated on the `ob` side and we'd see the changes in the returned `fr` version?

Reply:
it shouldn't matter, the ob object is parsed one time during the request and the data is converted to the FR object then stored in the database